### PR TITLE
Spark example improvements

### PR DESCRIPTION
**Pull request by assapin**
turned the spark code into more structured
used spark ml tools to normalize the data and transform the labels

updated the training code to also compute accuracy and do cross validation
updated the validation code to ensure accuracy is about 90% before promoting to production

### CODE REVIEW

**Comment by assapin** [comment]
@zionrubin @aviemzur

File: examples/spark-app-demo/k8s/liminal.yml [hunk 1]
```diff
@@ -52,20 +52,22 @@ pipelines:
         type: spark
         description: prepare the data for training
         application_arguments:
+          - '--input_uri'
```
Comment (zionrubin) [inline examples/spark-app-demo/k8s/liminal.yml:55]: application arguments can be a dict. you can do the following instead: 
```YAML
application_arguments:
  --input_uri: '{{input_root_dir}}data/iris.csv'
  --output_uri: '{{training_data_path}}'
```
the result will be the same.

File: examples/spark-app-demo/k8s/liminal.yml [hunk 2]
```diff
@@ -20,7 +20,7 @@ name: MyFirstLiminalSparkApp
 super: InfraSpark
```
Comment (zionrubin) [inline examples/spark-app-demo/k8s/liminal.yml:20]: Can you update the code snippet in the doc? (docs/getting-started/spark-app-demo/)

**Comment by lior-schachter** [comment]
LGTM

**Pull request merged**